esrally/utils/io.py

```diff
@@ -274,24 +285,54 @@ def decompress(zip_name, target_directory):
     this function.
     """
     path_without_extension, extension = splitext(zip_name)
-    filename = basename(path_without_extension)
     if extension == ".zip":
         _do_decompress(target_directory, zipfile.ZipFile(zip_name))
     elif extension == ".bz2":
-        _do_decompress_manually(target_directory, filename, bz2.open(zip_name))
+        decompressor_args = ["pbzip2", "-d", "-k", "-m10000", "-c"]
+        decompressor_lib = bz2.open
+        _do_decompress_manually(target_directory, zip_name, decompressor_args, decompressor_lib)
     elif extension == ".gz":
-        _do_decompress_manually(target_directory, filename, gzip.open(zip_name))
+        decompressor_args = ["pigz", "-d", "-k", "-c"]
+        decompressor_lib = gzip.open
+        _do_decompress_manually(target_directory, zip_name, decompressor_args, decompressor_lib)
     elif extension in [".tar", ".tar.gz", ".tgz", ".tar.bz2"]:
         _do_decompress(target_directory, tarfile.open(zip_name))
     else:
         raise RuntimeError("Unsupported file extension [%s]. Cannot decompress [%s]" % (extension, zip_name))
 
 
-def _do_decompress_manually(target_directory, filename, compressed_file):
+def _do_decompress_manually(target_directory, filename, decompressor_args, decompressor_lib):
+    decompressor_bin = decompressor_args[0]
+    base_path_without_extension = basename(splitext(filename)[0])
+
+    if is_executable(decompressor_bin):
+        if _do_decompress_manually_external(target_directory, filename, base_path_without_extension, decompressor_args):
+            return
+    else:
+        console.warn(f"{decompressor_bin} not found in PATH. Using standard library, decompression will take longer.")
```

Comment:
On the command line this looks like:

```
[INFO] Decompressing track data from [/Users/daniel/.rally/benchmarks/data/geonames/documents-2-1k.json.bz2] to [/Users/daniel/.rally/benchmarks/data/geonames/documents-2-1k.json] ... [WARNING] pbzip2 not found in PATH. Using standard library, decompression will take longer.
[OK]
```

This looks a bit odd and I wonder whether we should just write the output to the logs? Alternatively we could maybe offer a separate function that allows to check this upfront because with `http_logs` (multiple indices) this gets a bit noisy:

```
[INFO] Decompressing track data from [/Users/daniel/.rally/benchmarks/data/http_logs/documents-181998-1k.json.bz2] to [/Users/daniel/.rally/benchmarks/data/http_logs/documents-181998-1k.json] ... [WARNING] pbzip2 not found in PATH. Using standard library, decompression will take longer.
[OK]
[INFO] Preparing file offset table for [/Users/daniel/.rally/benchmarks/data/http_logs/documents-181998-1k.json] ... [OK]
[INFO] Decompressing track data from [/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json.bz2] to [/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json] ... [WARNING] pbzip2 not found in PATH. Using standard library, decompression will take longer.
[OK]
[INFO] Preparing file offset table for [/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json] ... [OK]
[INFO] Decompressing track data from [/Users/daniel/.rally/benchmarks/data/http_logs/documents-201998-1k.json.bz2] to [/Users/daniel/.rally/benchmarks/data/http_logs/documents-201998-1k.json] ... [WARNING] pbzip2 not found in PATH. Using standard library, decompression will take longer.
[OK]
```

Reply:
Re: noise, however, this noise will only happen once; after the all corpora have been extracted it'll not be seen again.

Re: printed location of the warn, I initially thought about adding a newline. Then the final [OK] would land on the next line, if a warning gets emitted. I think refactoring the checks out of here is not worth the benefits. Maybe we can just stick to logging.

Reply:
I'm fine with just logging. :)

Reply:
Switched to logging only in https://github.com/elastic/rally/pull/947/commits/476e3fd9051fa3db43cf6e85ec9a29b0f704f3de